packages/react-components/src/form-label/form-label-counter.tsx

```diff
@@ -0,0 +1,63 @@
+import { ReactNode } from 'react';
+import { Flex } from '../flex';
+import { InputProps } from '../input';
+import { Text } from '../text';
+
+export const FormLabelCounter = ({
+  counterSideElement,
+  showCount,
+  mergedValue,
+  maxLength,
+  label,
+  inputId
+}: {
+  showCount?: boolean;
+  counterSideElement?: ReactNode;
+  mergedValue?: InputProps['value'];
+  maxLength?: number;
+  label?: string;
+  inputId?: string;
+}) => {
+  if (!showCount && !counterSideElement) return null;
+  return (
+    <FormLabelCounterWrapper
+      hasCounterSideElement={!!counterSideElement}
+      showCount={showCount}
+    >
+      {showCount && (
+        <Text
+          css={{
+            color: '$neutral700',
+            ...(!label ? { marginLeft: 'auto' } : {})
+          }}
+          id={`${inputId}-char-count`}
+          data-testid="form-label-counter"
+        >
+          {mergedValue?.toString().length || 0}
+          {maxLength ? `/${maxLength}` : null}
+        </Text>
+      )}
+      {counterSideElement || null}
```

Comment:
I feel like we should do a simple `children` here rather than bringing this prop. 
in terms of output, I understand its same, but the prop name is a bit edgy.